apisix/core/request.lua

```diff
@@ -223,7 +223,7 @@ function _M.get_body(max_size, ctx)
 
     local file_name = req_get_body_file()
     if not file_name then
-        return nil
+        return nil, "request body has zero size"
```

Comment:
We can't do that as it will break ext-plugin, FaaS, and other plugins. What about  changing the graphQL code like that: https://github.com/apache/apisix/blob/e9fab8b2e5b5cb79a302faf8b5c9752766681c63/apisix/plugins/request-validation.lua#L83

Reply:
I also found this issue yesterday, so was wondering if it is possible to add a request method check at the entry of the get_body function. When the request is not `POST` or `PUT`, the operation is returned directly. Respond with proper error messages when `if not file_name then`.

This not only follows the rfc7231 specification while avoiding unnecessary body reading, but is also compatible with existing code.

like this:
```lua
local function check_get_body_method()
    -- https://datatracker.ietf.org/doc/html/rfc7231#section-4.3
    local method = req_get_method()
    method = str_upper(method)
    if method ~= "post" or method ~= "put" then
        log.info("only `POST` and `PUT` requests are supported to get the body, ",
                 "the current request method: ", method)
        return false
    end

    return true
end


function _M.get_body(max_size, ctx)
    if not check_get_body_method() then
        return
    end

	...

	local file_name = req_get_body_file()
    if not file_name then
        return nil, "request body has zero size"
    end
	
	...
end
```

Reply:
I don't think it is a good idea to mess the request body with the request method. Actually, you can send a request body even when the method is `GET`.

Reply:
I understand that any method can send the request body, but this is not RFC-compliant.